inference-engine/src/mkldnn_plugin/nodes/mkldnn_transpose_node.cpp

```diff
@@ -148,508 +156,123 @@ void MKLDNNTransposeNode::createPrimitive() {
-            return srcMemPtr->GetDesc().isPlainFormat() && MB == srcMemPtr->GetDims()[0];
-        })},
-        {{0, 2, 1, 3, 4}, MKLDNNTransposeNode::TransposeImpl(transpose_to_02134, [](int MB, MKLDNNMemoryPtr& srcMemPtr, MKLDNNMemoryPtr& dstMemPtr) {
-            return srcMemPtr->GetDesc().isPlainFormat();
-        })},
-        {{0, 2, 4, 3, 1}, MKLDNNTransposeNode::TransposeImpl(transpose_to_02431, [](int MB, MKLDNNMemoryPtr& srcMemPtr, MKLDNNMemoryPtr& dstMemPtr) {
-            return srcMemPtr->GetDesc().isPlainFormat();
-        })},
-        {{0, 4, 2, 3, 1}, MKLDNNTransposeNode::TransposeImpl(transpose_to_04231, [](int MB, MKLDNNMemoryPtr& srcMemPtr, MKLDNNMemoryPtr& dstMemPtr) {
-            return srcMemPtr->GetDesc().isPlainFormat();
-        })},
-        {{0, 3, 1, 2}, MKLDNNTransposeNode::TransposeImpl(transpose_to_0312, [](int MB, MKLDNNMemoryPtr& srcMemPtr, MKLDNNMemoryPtr& dstMemPtr) {
-            return srcMemPtr->GetDesc().isPlainFormat();
-        })},
-        {{1, 0, 2}, MKLDNNTransposeNode::TransposeImpl(transpose_to_102, [](int MB, MKLDNNMemoryPtr& srcMemPtr, MKLDNNMemoryPtr& dstMemPtr) {
-            return srcMemPtr->GetDesc().isPlainFormat() && MB == srcMemPtr->GetDims()[0];
-        })},
-        {{0, 2, 3, 4, 1}, MKLDNNTransposeNode::TransposeImpl(transpose_to_02341, [](int MB, MKLDNNMemoryPtr& srcMemPtr, MKLDNNMemoryPtr& dstMemPtr) {
-            return srcMemPtr->GetDesc().isPlainFormat();
-        })},
-        {{0, 4, 1, 2, 3}, MKLDNNTransposeNode::TransposeImpl(transpose_to_04123, [](int MB, MKLDNNMemoryPtr& srcMemPtr, MKLDNNMemoryPtr& dstMemPtr) {
-            return srcMemPtr->GetDesc().isPlainFormat();
-        })},
-};
-
 void MKLDNNTransposeNode::execute(mkldnn::stream strm) {
     auto &dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
     auto &srcMemPtr = getParentEdgeAt(0)->getMemoryPtr();
     int MB = batchToProcess();
 
-    if (prec == Precision::FP32 && !getParentEdgeAt(0)->getMemory().GetDesc().isTailCFormat()) {
-        for (const auto &impl : OptimizedCases) {
-            if (impl.first == order && impl.second.isValidParams(MB, srcMemPtr, dstMemPtr)) {
-                impl.second.execute(MB, srcMemPtr, dstMemPtr);
-                return;
-            }
-        }
+    if (isOptimized) {
+        const auto precision = getParentEdgeAt(0)->getDesc().getPrecision();
+        TransposeContext ctx = {this, srcMemPtr, dstMemPtr, MB};
+        OV_SWITCH(MKLDNNPlugin, TransposeOptimizedEmitter, ctx, precision,
+                  OV_CASE(InferenceEngine::Precision::FP32, float),
+                  OV_CASE(InferenceEngine::Precision::I32, int32_t),
+                  OV_CASE(InferenceEngine::Precision::BF16, bfloat16_t),
+                  OV_CASE(InferenceEngine::Precision::I8, int8_t),
+                  OV_CASE(InferenceEngine::Precision::U8, uint8_t));
```

Comment:
Why do you need to separate cases U8/I8 and FP32/I32 for data movement operation? Please fix in separate PR.

Reply:
Good point! Thank you
Separate [PR#5814](https://github.com/openvinotoolkit/openvino/pull/5814)